python_modules/dagster/dagster/core/storage/migration/utils.py

```diff
@@ -259,3 +260,32 @@ def create_schedule_secondary_index_table():
             db.Column("create_timestamp", db.DateTime, server_default=db.text("CURRENT_TIMESTAMP")),
             db.Column("migration_completed", db.DateTime),
         )
+
+
+def create_instigators_table():
+    if not has_table("instigators") and not has_table("jobs"):
+        # not a schedule storage db
+        return
+
+    if (
+        has_table("instigators")
+        and has_column("job_ticks", "selector_id")
+        and has_column("jobs", "selector_id")
+    ):
+        # already migrated
+        return
+
+    op.create_table(
+        "instigators",
+        db.Column("id", db.Integer, primary_key=True, autoincrement=True),
+        db.Column("selector_id", db.String(255), unique=True),
+        db.Column("repository_name", db.Text),
+        db.Column("status", db.String(63)),
+        db.Column("instigator_type", db.String(63), index=True),
+        db.Column("instigator_body", db.Text),
+        db.Column("create_timestamp", db.DateTime, server_default=get_current_timestamp()),
+        db.Column("update_timestamp", db.DateTime, server_default=get_current_timestamp()),
+    )
+
+    op.add_column("jobs", db.Column("selector_id", db.String(255)))
```

Comment:
So, we need this because we need to know when to delete from the `instigators` table.  I think I erred on the side of being very conservative... if there are two origin_ids with the same selector and I delete the unreachable schedule, then I might accidentaly delete from the instigator table the active one.

Instead, we're only deleting from this instigator table if there are no matching selectors.  This requires that the selector_id is populated in the `jobs` table, which is why it's part of the "required" data migration (which happens right after the schema is migrated).